### PR TITLE
Fixed how things get space cash worth, and richest escapee scoreboard line

### LINKED ISSUE #1784: End of round stats not displaying richest shuttle escapee correctly
Yesterday, while playing, I had gotten a large sum of credits accumulated in my character ATM account thing, over 60,000. At the end of the round, I was listed as the richest escapee, but it showed that the amount I escaped with was 4,000. Is it maybe that the amount it can display caps at 4,000?
This is my first github post thing, sorry if it's bad/terrible.

### PR DESCRIPTION
**Pull request by Tastyfish**
This PR in general fixes how things get the credit value of space cash. Many machines, etc previously just used the `cash.worth` var, ignoring that space cash is a stack with an `amount` var as well. In particular, the ATM transaction log and the spacecash object code itself were mixing up the worth and the total value.
* There is now a `get_total()` proc that everything calls instead, abstracting out how the cash stack represents its value. 
* The scoreboard code for *richest escapee* has also been changed to actually check the id's associated bank account, rather than the phantom var `id.money`, which was literally only referenced at its definition and the scoreboard code. That var has now been removed. It now also handles containers more than one level and uses `get_total()`.

Fixes #1784

### CODE REVIEW

**Comment by Crazylemon64** [comment]
:money_with_wings:

File: code/modules/economy/cash.dm [hunk 1]
```diff
@@ -17,50 +17,34 @@
 	var/worth = 0
 
 /obj/item/weapon/spacecash/c1
-	name = "1 credip chip"
```
Comment (tigercat2000) [inline code/modules/economy/cash.dm]: this is kinda important..

Comment (Tastyfish) [inline code/modules/economy/cash.dm]: That special case misspelling was a feature?

Comment (tigercat2000) [inline code/modules/economy/cash.dm]: no, the spacecash having a name.

Comment (tigercat2000) [inline code/modules/economy/cash.dm]: Oh, I see, weapon/spacecash does magic on it. Why these both exist, I don't know.

**Comment by Spacemanspark** [comment]
http://www.terribleideas.org/gettemmoney/ Courtesy of Ambassador Magikarp.

**Comment by Tastyfish** [comment]
That link perfectly captures the round I had yesterday which led me to fix this. $20k uncounted on the scoreboard was a war crime.

File: code/modules/economy/cash.dm (hunk 1 above)
Comment (Tastyfish) [inline code/modules/economy/cash.dm]: Yeah, them being named manually was redundant with New() doing it for you (and replacing the names). I removed it from the individual definitions so there wasn't the impression that those assigned names were actually being used anywhere.

Comment (tigercat2000) [inline code/modules/economy/cash.dm]: I just realized how fucking pointless this entire file is, it's not actually compiled

Comment (Tastyfish) [inline code/modules/economy/cash.dm]: Oh yeah, didn't notice that it was excluded. Well then the changes in that file are meaningless anyway. (And that would explain it appearing to be super redundant).

**Comment by TheDZD** [comment]
@Tastyfish Ready to be merged then?

**Comment by Tastyfish** [comment]
Yes

**Pull request merged**